### PR TITLE
Rename master to aggregator

### LINKED ISSUE #238: rename master to controller
In the spirit of https://github.com/kubernetes/website/issues/6525 I'd like to rename master to controller.

This will take a bit of time, but all new things should refer to master as controller.

### PR DESCRIPTION
**Pull request by johnSchnake**
**What this PR does / why we need it**:
Slightly more intuitive and useful name. Kept the old name
as an alias which we can decide to remove later if we want.

Did not update a few of the config values that are named with
Master in order to avoid potentially breaking changes.

Also took this as an opportunity to slightly tweak the logic
of how the master command is run (utilizing RunE to return errors
via the cobra command flow). This is how we'd like to see the
commands structured in the future since it gets away from a global
value being used.

Note, this will be backward compatible but if someone with the
newer Sonobuoy CLI tries to target an older Sonobuoy image, then
we will fail since we provide the newer command. This is not
terribly blocking since the user just has to use an older version
of the Sonobuoy CLI.

**Which issue(s) this PR fixes**
Fixes #238

**Special notes for your reviewer**:
The although this is backward compatible (Older clients can call this image) we actually run into a problem where _this_ version couldn't call older versions. To avoid that we'd have to add logic about the `if version < x then use a different command`. I'd really prefer not to do that. Maybe we just phase this in and in this release just do dual support but default to the `master` command and then next release we can switch over? That way we don't have such a sharp, and short, break in compatibility?

**Release note**:
```
Sonobuoy `master` command renamed to `aggregator`.
```

### CODE REVIEW

File: docs/enhancements/v2-worker-master.md [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 ## How it works now 
 
-Sonobuoy uses a worker-master model, where a master delegates tasks to worker
+Sonobuoy uses a worker-master model, where an aggregator delegates tasks to worker
```
[the remark this answers is not in the document]
Comment (zubron) [reply docs/enhancements/v2-worker-master.md]: On further look, it seems like this file is no longer present in the repo now that the docs are versioned and the `docs` directory was moved. This `v2-worker-master.md` file exists up to the `v0.14.3` release then is absent in `v0.15.0` and `master`. Should it be there?

[the remark this answers is not in the document]
Comment (johnSchnake) [reply docs/enhancements/v2-worker-master.md]: No, we werent actually linking to enhancements on the site (they were more of internal, k8s KEP-like documents) so I just removed them from the docs so we had one less thing to version.

**Comment by johnSchnake** [comment]
@zubron but what did you think about this section of the special notes:

> although this is backward compatible (Older clients can call this image) we actually run into a problem where this version couldn't call older versions. To avoid that we'd have to add logic about the if version < x then use a different command. I'd really prefer not to do that. Maybe we just phase this in and in this release just do dual support but default to the master command and then next release we can switch over? That way we don't have such a sharp, and short, break in compatibility?

**Comment by zubron** [comment]
@johnSchnake I forgot to come back to that comment. I'm getting a bit mixed up in the flow here, sorry. In what situation would we have a newer client calling older versions? The newer client would generate a new config (which would include the newer image) but I guess it could use a config from a previous version which could reference an old image?

**Comment by johnSchnake** [comment]
Will avoid changing things like top-level script names/etc which could affect backward compatibility. Will update comments/variable names etc to be consistent with our current, more descriptive naming.

**Pull request closed without merging**